### PR TITLE
[sw/silicon_creator] Configure the bootstrap pin only if enabled in OTP

### PR DESCRIPTION
**Pull request by alphan**
This PR sets the value of `ROM_BOOTSTRAP_EN` in the RMA OTP image to `kHardenedBoolTrue` and updates the pinmux driver to configure the bootstrap pin only if enabled in OTP.

### CODE REVIEW

File: sw/device/silicon_creator/lib/drivers/pinmux.c [hunk 1]
```diff
@@ -47,32 +30,60 @@ typedef struct pinmux_output {
...
+    .pad = kTopEarlgreyPinmuxInselIoc3,
 };
 
+/**
+ * Bootstrap pin.
+ */
+static const pinmux_input_t kInputBootstrap = {
+    .periph = kTopEarlgreyPinmuxPeripheralInGpioGpio17,
+    .pad = kTopEarlgreyPinmuxInselIob8,
+};
```
Comment (cfrantz) [inline sw/device/silicon_creator/lib/drivers/pinmux.c:46]: TODO: the real chip will have 3 strapping pins encoding 64-possible values by way of weak and strong pull up/down resistors.  Configure the 3 pins as inputs.

File: sw/device/silicon_creator/lib/drivers/pinmux.c [hunk 2]
```diff
@@ -47,32 +30,60 @@ typedef struct pinmux_output {
...
-     */
-    {.pad = kTopEarlgreyPinmuxMioOutIoc4,
-     .periph = kTopEarlgreyPinmuxOutselUart0Tx},
+static const pinmux_input_t kInputUart = {
+    .periph = kTopEarlgreyPinmuxPeripheralInUart0Rx,
+    .pad = kTopEarlgreyPinmuxInselIoc3,
 };
 
+/**
+ * Bootstrap pin.
```
Comment (moidx) [inline sw/device/silicon_creator/lib/drivers/pinmux.c:41]: nit: Are we planning to enable support in the mask ROM to be able to decode all the bootstrap pins? Can we add a note to clarify?

File: sw/device/silicon_creator/lib/drivers/pinmux.c [hunk 3]
```diff
@@ -47,32 +30,60 @@ typedef struct pinmux_output {
...
-  const uint32_t kInputBase =
-      TOP_EARLGREY_PINMUX_AON_BASE_ADDR + PINMUX_MIO_PERIPH_INSEL_0_REG_OFFSET;
-  for (uint32_t i = 0; i < ARRAYSIZE(kPinmuxInputs); ++i) {
-    uint32_t reg = kPinmuxInputs[i].periph * sizeof(uint32_t);
-    uint32_t val = kPinmuxInputs[i].pad;
-    abs_mmio_write32(kInputBase + reg, val);
+  uint32_t bootstrap_en = otp_read32(OTP_CTRL_PARAM_ROM_BOOTSTRAP_EN_OFFSET);
+  if (launder32(bootstrap_en) == kHardenedBoolTrue) {
+    HARDENED_CHECK_EQ(bootstrap_en, kHardenedBoolTrue);
+    configure_input(kInputBootstrap);
```
Comment (cfrantz) [inline sw/device/silicon_creator/lib/drivers/pinmux.c:84]: Should we make configuring the pins conditional on the OTP word or should we unconditionally configure the strapping pins as inputs and use the OTP word to control entry into bootstrap mode?

ROM_EXT and owner code are probably going to want to read the straps too.

**Review by moidx — APPROVED** [review]

File: sw/device/silicon_creator/lib/drivers/pinmux.c (hunk 3 above)
Comment (alphan) [reply sw/device/silicon_creator/lib/drivers/pinmux.c:84]: I was planning to control both (pin configuration and bootstrap entry) using the OTP value (no need to configure the pins if we are not going to use them) but can revert this if you think we should configure the pins. @moidx any thoughts?

**Review by mcy — APPROVED** [review]

File: sw/device/silicon_creator/lib/drivers/pinmux.c (hunk 1 above)
Comment (alphan) [reply sw/device/silicon_creator/lib/drivers/pinmux.c]: Created #11934

File: sw/device/silicon_creator/lib/drivers/pinmux.c (hunk 2 above)
Comment (alphan) [reply sw/device/silicon_creator/lib/drivers/pinmux.c:41]: Captured in #11934, added TODO

**Review by cfrantz — APPROVED** [review]

**Pull request merged**